recommend/policyRules.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package recommend
+
+import (
+	_ "embed" // need for embedding
+	"errors"
+
+	"github.com/clarketm/json"
+
+	"github.com/fatih/color"
+	log "github.com/sirupsen/logrus"
+)
+
+// MatchSpec spec to match for defining policy
+type MatchSpec struct {
+	Name         string      `json:"name"`
+	Precondition string      `json:"precondition"`
+	Description  Description `json:"description"`
+	Rules        Rules       `json:"rules"`
+	OnEvent      OnEvent     `json:"onEvent"`
+}
+
+// Ref for the policy rules
+type Ref struct {
+	Name string   `json:"name"`
+	URL  []string `json:"url"`
+}
+
+// Description detailed description for the policy rule
+type Description struct {
+	Refs     []Ref  `json:"refs"`
+	Tldr     string `json:"tldr"`
+	Detailed string `json:"detailed"`
+}
+
+// PathRule specifics for the path/dir rule. Note that if the Path ends in "/" it is considered to be Directory rule
+type PathRule struct {
+	FromSource string   `json:"fromSource"`
+	Path       []string `json:"path"`
+	Recursive  bool     `json:"recursive"`
+	Owneronly  bool     `json:"owneronly"`
+}
+
+// Rules set of applicable rules. In the future, we might have other types of rules.
+type Rules struct {
+	PathRule PathRule `json:"pathRule"`
```

Comment:
Will we handle addition of Process/File/Network in a subsequent PR?

Reply:
Yes, there is already a discussion on this. We will need this immediately but we will take care in a separate PR.